CMakeLists.txt

```diff
@@ -32,72 +33,108 @@ include_directories(ffig/include)
-execute_process(COMMAND luajit -e "print('testing')" RESULT_VARIABLE result OUTPUT_QUIET ERROR_QUIET)
+# FIXME: Make this less hacky, ideally using a real CMake module
+execute_process(COMMAND luajit -e "print('testing')" 
+                RESULT_VARIABLE result 
+                OUTPUT_QUIET 
+                ERROR_QUIET)
 if(result)
   message(STATUS "Lua tests disabled: luajit cannot execute a simple Lua script.")
 else()
   set(LuaJIT_FOUND 1)
   set(LuaJIT_EXECUTABLE "luajit")
+  # FIXME: Get Lua bindings working well enough to work with Shape classes and 
+  # add LUA to all_ffig_bindings.
 endif()
 
 # Look for C#
 # FIXME: Make this less hacky, ideally using a real CMake module
-execute_process(COMMAND dotnet --version RESULT_VARIABLE result OUTPUT_QUIET ERROR_QUIET)
+execute_process(COMMAND dotnet --version 
+                RESULT_VARIABLE result 
+                OUTPUT_QUIET 
+                ERROR_QUIET)
 if(result)
   message(STATUS "Dotnet tests disabled: dotnet executable not found.")
 else()
   set(dotnet_FOUND 1)
-  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/TestTree.cs.in ${CMAKE_CURRENT_BINARY_DIR}/generated/TestTree.cs)
-  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/TestShape.cs.in ${CMAKE_CURRENT_BINARY_DIR}/generated/TestShape.cs)
-  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/ffig.net.csproj.in ${CMAKE_CURRENT_BINARY_DIR}/generated/ffig.net.csproj)
+  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/TestTree.cs.in 
+                 ${CMAKE_CURRENT_BINARY_DIR}/generated/TestTree.cs)
+  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/TestShape.cs.in 
+                 ${CMAKE_CURRENT_BINARY_DIR}/generated/TestShape.cs)
+  configure_file(${CMAKE_CURRENT_LIST_DIR}/tests/dotnet/ffig.net.csproj.in 
+                 ${CMAKE_CURRENT_BINARY_DIR}/generated/ffig.net.csproj)
+  list(APPEND all_ffig_bindings "DOTNET")
 endif()
 
-set(all_ffig_bindings "PYTHON;CPP;CPP_MOCKS;DOTNET")
-if(Ruby_FOUND)
-  list(APPEND all_ffig_bindings "RUBY")
-endif()
-
-if(Go_FOUND)
-  list(APPEND all_ffig_bindings "GO")
-endif()
 
+# Set FFIG_SOURCE so that FFIG-targets are sensitive to changes in FFIG code and templates.
+# The variable FFIG_SOURCE is used by ffig.cmake and is only needed when developing FFIG.
 file(GLOB_RECURSE FFIG_SOURCE ${CMAKE_CURRENT_LIST_DIR}/ffig/*)
 
 # Add FFIG build targets
-ffig_add_library(NAME Shape INPUTS tests/input/Shape.h ${all_ffig_bindings})
-ffig_add_library(NAME Tree INPUTS tests/input/Tree.h NOEXCEPT PYTHON CPP D SWIFT DOTNET)
-ffig_add_library(NAME Asset INPUTS tests/input/Asset.h PYTHON LUA)
-ffig_add_library(NAME Animal INPUTS tests/input/Animal.h PYTHON)
+ffig_add_library(NAME Shape 
+                 INPUTS tests/input/Shape.h 
+                 ${all_ffig_bindings})
+
+ffig_add_library(NAME Tree 
+                 INPUTS tests/input/Tree.h 
+                 CPP 
+                 D 
+                 DOTNET
+                 NOEXCEPT 
```

Comment:
I find it slightly odd that the `NOEXCEPT` option is mixed in with a bunch of language output options. I wonder if we'd be better served by a `ffig_add_library_noexcept`. Or maybe an explicit `NOEXCEPT [True|False|Deduced]`?

Reply:
Can we defer making this change to a follow up PR? Ideally this PR is mechanical cleanup.

Reply:
Sure.